tests/Doctrine/Tests/ORM/Functional/EnumTest.php

```diff
@@ -58,6 +58,27 @@ public function testEnumMapping(string $cardClass): void
         $this->assertEquals(Suit::Clubs, $fetchedCard->suit);
     }
 
+    public function testFindByEnum(): void
+    {
+        $this->setUpEntitySchema([Card::class]);
+
+        $card1       = new Card();
+        $card1->suit = Suit::Clubs;
+        $card2       = new Card();
+        $card2->suit = Suit::Hearts;
+
+        $this->_em->persist($card1);
+        $this->_em->persist($card2);
+        $this->_em->flush();
+
+        unset($card1, $card2);
+        $this->_em->clear();
+
+        foreach ($this->_em->getRepository(Card::class)->findBy(['suit' => Suit::Clubs]) as $card) {
+            $this->assertSame(Suit::Clubs, $card->suit);
+        }
```

Comment:
If `findBy()` returns `[]`, this test will pass. Let's assert the size of the result?